packages/arc-auth/cdk/src/env.ts

```diff
@@ -0,0 +1,81 @@
+import {DataAwsSecurityGroup} from '@cdktf/provider-aws/lib/data-aws-security-group';
+import {DataAwsSsmParameter} from '@cdktf/provider-aws/lib/data-aws-ssm-parameter';
+import {DataAwsSubnets} from '@cdktf/provider-aws/lib/data-aws-subnets';
+import {TerraformStack} from 'cdktf';
+import {readFileSync} from 'fs';
+
+export const env = {
```

Comment:
Why do we need to add properties here also?
This would be duplication of effort for the user whenever user adds a new property. User will have to add that property here also.
We can just pick the properties from the .env. Whatever properties are there we can fetch from ssm and feed it into lambda env variables.
cc: @jamescrowley321

Reply:
These are all params that should be loaded from SSM.

The idea is that the env file is a list of SSM param keys, we query SSM for those keys at runtime, and then we pass the values from SSM to the CDK where needed.

Reply:
we will be maintainig all our env variables here only. not in .env.
we can bring our values from .env , but in thatcase we can not use process.env for that. either we have to use fs methods or something else. because process.env brings in many other variables too. And using fs methods with .env, it does not seems usual, i also went through some workaround but didn't get the funcitonal way. that's why i followed this approach.

storing all our env inside an object makes them accessible easily, we don't have to add any further logic to separate our required env from the system env variables.

this is one approcah, and may be its possible to do in some better way

Reply:
Let's discuss it tomorrow @Tyagi-Sunny

Reply:
@jamescrowley321 so we can easily assume that the properties that user wants to fetch from SSM. User will have to specify those beforehand.
So we can create an array propertiesToFetch and user have to add properties there
whatever properties user will add we'll fetch it from ssm.

Reply:
Here is what I recommend
- ENV file is a list of values like `DB_HOST_SSM_PARAM_KEY=whatever/the/key/value/is`
- Create a configuration type, i.e. `AuthServiceConfiguration`
- Create a type such as `AuthServiceConfigurationLoader` (or w/e you want to name it) and a concrete implementation that encapsulates reading the ENV variables and looking up the values from SSM to return an instance of `AuthServiceConfiguration`.
- Only reference `AuthServiceConfiguration`
- Inject `AuthServiceConfigurationLoader` where you can
 - Then the devs can also use that interface to create a Mock/Stub/Local version of the `AuthServiceConfigurationLoader` to read from other param stores
 - We may not always use SSM to load the variables, so decouple things where you can.

Reply:
Why cant we do all of this in Code Pipeline ? Ideally we do these things in code pipeline or jenkins or whatever tool we use for CI/CD. So that application code remains unaware of tool to use. It just accesses everything from env. CI/CD pipeline scripts read values from SSM or Vault and put into the env while deploying the service.